tests/manage/pv_services/test_resource_deletion_during_pvc_pod_creation_and_io.py

```diff
@@ -0,0 +1,275 @@
+        )
+
+        # Updating self.pod_objs for the purpose of teardown
+        self.pod_objs.extend(io_pods)
+
+        # Do setup for running IO on pods
+        log.info("Setting up pods for running IO")
+        for pod_obj in io_pods:
+            pod_obj.workload_setup(storage_type='fs')
+        log.info("Setup for running IO is completed on pods")
+
+        # Start creating new pods
+        log.info("Start creating new pods.")
+        bulk_pod_create = executor.submit(
+            helpers.create_pods, pvc_objs_list=pvc_objs_new_pods,
+            interface_type=self.interface, wait=False,
+            namespace=self.namespace
+        )
+
+        # Start creation of new PVCs
+        log.info("Start creating new PVCs.")
+        bulk_pvc_create = executor.submit(
+            helpers.create_multiple_pvcs, sc_name=self.sc_obj.name,
+            namespace=self.namespace, number_of_pvc=self.num_of_new_pvcs,
+            size=self.pvc_size
+        )
+
+        # Start IO on each pod
+        log.info("Start IO on pods")
+        for pod_obj in io_pods:
+            pod_obj.run_io(storage_type='fs', size=f'{self.pvc_size_int - 1}G')
+        log.info("IO started on all pods.")
+
+        # Delete the resource
+        DISRUPTION_OPS.delete_resource()
+
+        # Getting result of PVC creation as list of PVC objects
+        pvc_objs_new = bulk_pvc_create.result()
+
+        # Updating self.pvc_objs_new for the purpose of teardown
+        self.pvc_objs_new.extend(pvc_objs_new)
+
+        # Verification of PVC status is done in 'create_multiple_pvcs' function
+        log.info("Verified: New PVCs are Bound.")
+
+        # Getting result of pods creation as list of Pod objects
+        pod_objs_new = bulk_pod_create.result()
+
+        # Updating self.pod_objs for the purpose of teardown
+        self.pod_objs.extend(pod_objs_new)
+
+        # Verify new pods are Running
+        for pod_obj in pod_objs_new:
+            assert pod_obj.ocp.wait_for_resource(
+                condition=constants.STATUS_RUNNING,
+                resource_name=pod_obj.name, timeout=120
+            ), (
+                f"Wait timeout: Pod {pod_obj.name} is not in 'Running' "
+                f"state even after 120 seconds."
+            )
+        log.info("Verified: All pods are Running.")
+
+        # Verify IO
+        log.info("Fetching IO results.")
+        for pod_obj in io_pods:
+            fio_result = pod_obj.get_fio_results()
+            log.info(f"IOPs after IO on pod {pod_obj.name}:")
+            log.info(
+                f"Read: {fio_result.get('jobs')[0].get('read').get('iops')}"
+            )
+            log.info(
+                f"Write: {fio_result.get('jobs')[0].get('write').get('iops')}"
+            )
+        log.info("Verified IO result on pods.")
+
+        # Delete pods
+        all_pod_objs = io_pods + pod_objs_new
+        for pod_obj in all_pod_objs:
+            pod_obj.delete(wait=True)
```

Comment:
don't wait on each delete.
Delete all and after that, loop over all pods again with wait_for_delete() on each

Reply:
Done